lib/spack/spack/fetch_strategy.py

```diff
@@ -297,51 +283,63 @@ def mirror_id(self):
         return os.path.sep.join(
             ['archive', self.digest[:2], self.digest])
 
+    @property
+    def candidate_urls(self):
+        return [self.url] + (self.mirrors or [])
+
     @_needs_stage
     def fetch(self):
         if self.archive_file:
             tty.msg("Already downloaded %s" % self.archive_file)
             return
 
+        for url in self.candidate_urls:
+            try:
+                partial_file, save_file = self._fetch_from_url(url)
+                if save_file:
+                    os.rename(partial_file, save_file)
+                break
+            except FetchError as e:
+                tty.msg(str(e))
+                pass
+
+        if not self.archive_file:
+            raise FailedDownloadError(self.url)
+
+    def _fetch_from_url(self, url):
         save_file = None
         partial_file = None
         if self.stage.save_filename:
             save_file = self.stage.save_filename
             partial_file = self.stage.save_filename + '.part'
-
-        tty.msg("Fetching %s" % self.url)
-
+        tty.msg("Fetching %s" % url)
         if partial_file:
             save_args = ['-C',
                          '-',  # continue partial downloads
                          '-o',
                          partial_file]  # use a .part file
         else:
             save_args = ['-O']
-
         curl_args = save_args + [
             '-f',  # fail on >400 errors
             '-D',
             '-',  # print out HTML headers
             '-L',  # resolve 3xx redirects
-            self.url,
+            # Timeout if can't establish a connection after 10 sec.
+            '--connect-timeout', '10',
```

Comment:
Just to make sure, I checked and this option is supported since 19 years ago.  It's also in RHEL 6.  😄